### ISSUE TITLE
Constructor does not raise error when typo in parameters

### ISSUE THREAD

**Issue by garandria** [pysathq/pysat#39]
The solver's constructor `Solver` does not raise any error in case of typo or parameters that does not even exist in the code. For instance:
```python
s = Solver(boostrp_with=cnf.clauses)
```
works fine and the solver seems built. (Notice that I wrote `bosstrp_with` instead of `bootstrap_with').
It would be great if we are aware of such errors because we can work on an obsolete solver.

For example:
```python
class test:
    def __init__(self, param1=None, param2=None):
        pass
```
we get an error from Python when:
```python
>>> t = test(param3='string')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
TypeError: __init__() got an unexpected keyword argument 'param3'
```

Thank you.

**Comment by alexeyignatiev** [comment]
Hi @garandria,

Thanks a lot for reporting. This is a bug and it is fixed in the new version (`0.1.5.dev6`). Please, check. I am closing the issue now but if the problem remains, feel free to reopen.

Alexey

**Comment by garandria** [comment]
Thanks for your anser @alexeyignatiev  .

I would like to know why do you need the `**kwargs` parameter in the
basic constructor https://github.com/pysathq/pysat/blob/8ab8b514136f7578888cf32168fb74541ea874a4/pysat/solvers.py#L305 and the `new` function
https://github.com/pysathq/pysat/blob/8ab8b514136f7578888cf32168fb74541ea874a4/pysat/solvers.py#L328 whereas it is not used anywhere ?
